ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,102 @@
+# 39. Workspace Deprecation
+
+Some of the text was taken from - https://github.com/konflux-ci/architecture/pull/187
+
+## Status
+
+WiP
+
+## Context
+
+The purpose of this ADR is to revisit the workspace concept, understand the purpose
+of it, and offer an alternative implementation based on native Kubernetes APIs
+and other successful cloud native open source projects.
+
+We inherited the "workspace" concept from our early days when we expected to be integrated with the [KCP] project. Long ago (over a year at the time of this writing), we decided to remove [KCP] from the architecture of the Konflux project.
+
+In order to support multi-cluster deployment, [kubesaw] was used instead of KCP.
+
+Today, Workspaces are an important concept in the Konflux user experience. The user onboards into a workspace of their own. A team member of theirs may add them to a team workspace. As a platform administrator, if you hear about a user having trouble, you'll ask them what workspace they're working in.
+
+A Workspace is an abstraction of a namespace. Each workspace has a single namespace
+which backs it.
+
+Workspaces are required by [kubesaw] for providing a transparent multi-cluster communication for us. A user's workspace may be allocated to one cluster or another - but our proxy layer keeps the user blissfully unaware of this. Workspaces can be placed on clusters either automatically (using [kubesaw] mechanisms) or manually by a platform administrator.
+
+Workspace roles give a way to describe who has what rights in the workspace.
+
+Workspaces happen to be our mechanism for managing quota - a fact inherited from the [kubesaw] stack.
+
+### Some changes
+
+Some relevant changes that have occurred or are ongoing.
+
+* We want to make Konflux a successful open source projects that fits into the
+cloud native apps ecosystem. We want everything to work on vanilla Kubernetes - no OpenShift required.
+
+* We had our [first upstream Konflux release](https://github.com/konflux-ci/konflux-ci) which doesn't use [kubesaw] or relying
+on the Workspace abstraction, and is deployable on vanilla Kubernetes.
+
+* We made a decision (now long ago) to drop [KCP] from our architecture, which initially introduced the workspace concept. Both transparent multi-cluster as a feature and workspaces as a concept came from KCP.
```

Comment:
Just to be clear, mutli-cluster concept that is in KubeSaw was designed and implemented many years before kcp project even started - in 2019. 
Support for the multi-cluster architecture was implemented to solve real problems with scaling and with the real limit for the number of namespaces per cluster. We faced this problem problem before (in a previous project), that's why we designed KubeSaw as a multi-cluster solution from the very beginning and it has paid off many times. We also know that other RH customers are facing the same scaling problem in similar (pipeline) environments that rely on one cluster only. Apart from that, KubeSaw works in a single-cluster environment without any problems.

As for the workspace concept. As you may remember, workspace concept in kcp was something completely different. The only common thing is the name "workspace"; otherwise, the concept, abstraction, logic, etc... is completely different. Workspace API is just a public API for internal Space abstraction, which was implemented before kcp - in 2021.

Reply:
I accepted your suggestion for those lines.